website/static/js/filepage/revisions.js

```diff
@@ -145,6 +147,57 @@ var FileRevisionsTable = {
             ].filter(TRUTHY));
         };
 
+        self.tagsInput = function() {
+            var guid = self.file.file_guid;
+            var name = self.file.name;
+            // Tags input
+            $('#node-tags').tagsInput({
+                interactive: self.canEdit(),
+                maxChars: 128,
+                onAddTag: function(tag){
+                    var url = nodeApiUrl + 'file/tags/' + guid + '/';
+                    var request = $.ajax({
+                        url: url,
+                        type: 'POST',
+                        contentType: 'application/json',
+                        dataType: 'JSON',
+                        data: JSON.stringify({tag: tag, fileName: name})
+                    });
+                    request.fail(function(xhr, textStatus, error) {
+                        Raven.captureMessage('Failed to add tag', {
```

Comment:
Since it is using raven capture here, once the request failed, user won't get straight forwards error message unless they open console.